app/src/main/java/org/mozilla/fenix/settings/CookieBannersFragment.kt

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings
+
+import android.os.Bundle
+import androidx.preference.Preference
+import androidx.preference.PreferenceFragmentCompat
+import androidx.preference.SwitchPreference
+import mozilla.components.concept.engine.Settings
+import org.mozilla.fenix.GleanMetrics.CookieBanners
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.ext.showToolbar
+
+/**
+ * Lets the user set up the cookie banners handling preferences.
+ */
+class CookieBannersFragment : PreferenceFragmentCompat() {
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        setPreferencesFromResource(R.xml.cookie_banner_preferences, rootKey)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        showToolbar(getString(R.string.preferences_cookie_banner_reduction))
+        setupPreferences()
```

Comment:
setupPreferences() should be called in onCreatePreferences after setPreferencesFromResource because onResume() gets called multiple times .